File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length == 0) {
            return 0;
        }
        Arrays.sort(nums);

        int consecutiveCount = 1;
        int lastGoodStreak = 1;
        for(int i=1; i<nums.length; i++) {
            if(nums[i-1] == nums[i]) {
                continue;
            }
            if(nums[i-1] + 1 == nums[i]) {
                consecutiveCount++;
            } else {
                consecutiveCount = 1;
            }
            lastGoodStreak = Math.max(lastGoodStreak, consecutiveCount);
        }
        return lastGoodStreak;
    }
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    lengthOfLIS(nums) {
        const lisCache = new Array(nums.length).fill(1);
        for(let i = nums.length - 1; i >= 0; i--) {
            for(let j = i + 1; j <=nums.length; j++) {
                if(nums[j] > nums[i]) {
                    lisCache[i] = Math.max(lisCache[i], 1 + lisCache[j]);
                }
            }
        }
        return Math.max(...lisCache);
    }
}
